### PR TITLE
feat: delete table

### LINKED ISSUE #121: API: delete a Table

### PR DESCRIPTION
**Pull request by lucafurrer**
Adds an API point to delete a table

Tables will deleted, when a project is deleted

fix #121

### CODE REVIEW

File: apis/core/hydra/table.ttl [hunk 1]
```diff
@@ -45,6 +45,24 @@ cc:Table
       hydra:property
         cc:csvw ;
     ] ;
+  hydra:supportedOperation
+    [
+      a
+        hydra:Operation,
+        schema:DeleteAction ;
+      hydra:title
+        "Delete Project" ;
```
Comment (martinmaillard) [inline apis/core/hydra/table.ttl:54]: Should probably be "Delete table"

**Review by martinmaillard — APPROVED** [review]

File: apis/core/lib/domain/csv-mapping/delete.ts [hunk 2]
```diff
@@ -21,6 +21,13 @@ export async function deleteMapping(csvMapping: NamedNode, store: ResourceStore)
     store.delete(sourceCollection)
   }
 
+  const tables = getTablesFromMapping(csvMapping)
+  for await (const table of tables) {
+    if (table.termType === 'NamedNode') {
+      await deleteTableWithoutSave(table, store)
+    }
```
Comment (tpluscode) [inline apis/core/lib/domain/csv-mapping/delete.ts:28]: I'd move the check inside the delete function (no-op if not named node)

```suggestion
    await deleteTableWithoutSave(table, store)
```

The parameter would become `Term`

File: apis/core/lib/domain/queries/table.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { NamedNode } from 'rdf-js'
+import { SELECT } from '@tpluscode/sparql-builder'
+
+import { parsingClient } from '../../query-client'
+import { cc } from '@cube-creator/core/namespace'
+
+export async function * getTablesFromMapping(csvMapping: NamedNode, client = parsingClient) {
```
Comment (tpluscode) [inline apis/core/lib/domain/queries/table.ts:7]: A tiny rename maybe?

```suggestion
export async function * getTablesForMapping(csvMapping: NamedNode, client = parsingClient) {
```

File: apis/core/lib/domain/table/delete.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+export async function deleteTableWithoutSave(resource: NamedNode<string>, store: ResourceStore): Promise<void> {
+  const table = await store.get(resource)
+  if (!table) return
+
+  // Delete in columnMappings
+  const columnMappings = table.out(cc.columnMapping).terms
+  for await (const columnMapping of columnMappings) {
+    if (columnMapping?.termType === 'NamedNode') {
+      await deleteColumnMapping(columnMapping, store)
+    }
```
Comment (tpluscode) [inline apis/core/lib/domain/table/delete.ts:30]: Ditto
```suggestion
    await deleteColumnMapping(columnMapping, store)
```

**Review by tpluscode — CHANGES_REQUESTED** [review]
Small changes + please fix what Martin pointed out

File: apis/core/hydra/table.ttl (hunk 1 above)
Comment (lucafurrer) [reply apis/core/hydra/table.ttl]: Yes

File: apis/core/lib/domain/csv-mapping/delete.ts (hunk 2 above)
Comment (lucafurrer) [reply apis/core/lib/domain/csv-mapping/delete.ts:28]: Should I do the same thing with the sources?

**Review by tpluscode — APPROVED** [review]

**Pull request merged**